mne/decoding/time_gen.py

```diff
@@ -776,16 +762,19 @@ def _predict(X, estimators):
     # Initialize results:
     n_epochs = X.shape[0]
     n_clf = len(estimators)
-    n_class = estimators[0].predict(X[0, :]).shape[-1]  # initialize
-    y_pred = np.ones((n_epochs, n_class, n_clf))
 
     # Compute prediction for each sub-estimator (i.e. per fold)
     # if independent, estimators = all folds
     for fold, clf in enumerate(estimators):
-        if n_class == 1:
-            y_pred[:, 0, fold] = clf.predict(X)
+        predict = clf.predict(X)
+        # initialize predict_results array
+        if fold == 0:
+            predict_size = predict.shape[1] if len(predict.shape) > 1 else 1
```

Comment:
len(predict.shape) -> predict.ndim

but I don't like the name predict for a y_pred...